sirepo/pkcli/job_agent.py

```diff
@@ -55,14 +57,49 @@ def default_command():
     pkdlog('{}', cfg)
     i = tornado.ioloop.IOLoop.current()
     d = _Dispatcher()
-    def s(n, x):
-        return i.add_callback_from_signal(d.terminate)
+    def s(*args):
+        return i.add_callback_from_signal(_terminate, d)
     signal.signal(signal.SIGTERM, s)
     signal.signal(signal.SIGINT, s)
     i.spawn_callback(d.loop)
     i.start()
 
 
+def start_sbatch():
+    def kill_agent(pid_file):
+        if socket.getfqdn() == pid_file.fqdn:
+            os.kill(pid_file.pid, signal.SIGKILL)
+        else:
+            try:
+                subprocess.check_output(
+                    ('ssh', pid_file.fqdn, 'kill', '-KILL', str(pid_file.pid)),
+                    stderr=subprocess.STDOUT,
+                )
+            except subprocess.CalledProcessError as e:
+                pkdp(
+                    'cmd={cmd} returncode={returncode} output={output}',
+                    **vars(e)
+                )
+    try:
+        f = pkio.py_path(_PID_FILE)
+        if f.exists():
+            kill_agent(pkjson.load_any(f))
+    except Exception as e:
+        pkdlog('error={} stack={}', e, pkdexc())
```

Comment:
Include `f` so we know what the error was about. It's arguable, but given there could be race conditions here, maybe better to just read the file, and catch the exception, and check if it was not_found ([EAFP](https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/)).

Reply:
Thanks for the EAFP link. I went that route. Look right?